core/src/main/scala/com/raphtory/api/algorithm/Multilayer.scala

```diff
@@ -0,0 +1,79 @@
+package com.raphtory.api.algorithm
+
+import com.raphtory.api.graphview.GraphPerspective
+import com.raphtory.api.graphview.MultilayerGraphPerspective
+import com.raphtory.api.graphview.ReducedGraphPerspective
+import com.raphtory.api.table.Row
+import com.raphtory.api.table.Table
+
+trait Multilayer extends BaseAlgorithm {
+  override type In  = MultilayerGraphPerspective
+  override type Out = MultilayerGraphPerspective
+
+  case class ChainedMultilayer(
+      first: Multilayer,
+      second: Generic
+  ) extends ChainedAlgorithm(first, second)
+          with Multilayer {
+
+    override def apply(graph: MultilayerGraphPerspective): graph.MultilayerGraph =
+      second(first(graph).clearMessages())
+
+    override def tabularise(graph: MultilayerGraphPerspective): Table =
+      second.tabularise(graph)
+  }
+
+  case class ChainedMultilayer2(
```

Comment:
I am not saying I have a better one, but we need to come up with a better name for this 😄